klippy/extras/rpi_temperature.py

```diff
@@ -0,0 +1,89 @@
+# Support for Raspberry Pi temperature sensor
+#
+# Copyright (C) 2020  Al Crate <al3ph@users.noreply.github.com>
+#
+# This file may be distributed under the terms of the GNU GPLv3 license.
+
+import logging
+
+RPI_REPORT_TIME = 1.0
+KELVIN_TO_CELSIUS = -273.15
+
+class RPiTemperature:
+    def __init__(self, config):
+        self.printer = config.get_printer()
+        self.name = config.get_name().split()[-1]
+        self.reactor = self.printer.get_reactor()
+
+        self.temp = 0.0
+        self.file_handle = None
+        self.min_temp = config.getfloat('min_temp', -40,
+                                        minval=KELVIN_TO_CELSIUS)
+        self.max_temp = config.getfloat('max_temp', 85,
+                                        above=self.min_temp)
+
+        self.printer.add_object("rpi_temperature " + self.name, self)
+        self.sample_timer = self.reactor.register_timer(
+            self._sample_pi_temperature
+        )
+        self.printer.register_event_handler(
+            "klippy:connect", self.handle_connect
+        )
+        self.printer.register_event_handler(
+            "klippy:disconnect", self.handle_disconnect
+        )
+
+    def handle_connect(self):
+        self.file_handle = open("/sys/class/thermal/thermal_zone0/temp", "r")
```

Comment:
This needs to have a try/except block to catch errors.

FWIW, I think it would be preferable to open the file in the `init()` method and throw a `config.error()` exception if the file can't be opened.